tests/spec/s-function.js

```diff
@@ -1,7 +1,177 @@
+            }).not.toThrow();
+            expect(function () {
+                noop.call(null, undefined, null, '1', 1, true, noop);
+            }).not.toThrow();
+            expect(function () {
+                noop.call('a', undefined, null, '1', 1, true, noop);
+            }).not.toThrow();
+            expect(function () {
+                noop.call(1, undefined, null, '1', 1, true, noop);
+            }).not.toThrow();
+            expect(function () {
+                noop.call(true, undefined, null, '1', 1, true, noop);
+            }).not.toThrow();
+            expect(function () {
+                noop.call(noop, undefined, null, '1', 1, true, noop);
+            }).not.toThrow();
+            expect(function () {
+                noop.call(new Date(), undefined, null, '1', 1, true, noop);
+            }).not.toThrow();
+        });
+        if (hasStrictMode) {
+            it('should have correct context in strict mode', function () {
+                'use strict';
+
+                var result;
+                var subject;
+                var testFn = function () {
+                    return this;
+                };
+                expect(function () {
+                    result = testFn.call();
+                }).not.toThrow();
+                expect(result).toBe(undefined);
+                expect(function () {
+                    result = testFn.call(undefined);
+                }).not.toThrow();
+                expect(result).toBe(undefined);
+                expect(function () {
+                    result = testFn.call(null);
+                }).not.toThrow();
+                expect(result).toBe(null);
+                expect(function () {
+                    result = testFn.call('a');
+                }).not.toThrow();
+                expect(result).toBe('a');
+                expect(function () {
+                    result = testFn.call(1);
+                }).not.toThrow();
+                expect(result).toBe(1);
+                expect(function () {
+                    result = testFn.call(true);
+                }).not.toThrow();
+                expect(result).toBe(true);
+                expect(function () {
+                    result = testFn.call(testFn);
+                }).not.toThrow();
+                expect(result).toBe(testFn);
+                subject = new Date();
+                expect(function () {
+                    result = testFn.call(subject);
+                }).not.toThrow();
+                expect(result).toBe(subject);
+            });
+        }
+        it('should have correct context in non-strict mode', function () {
+            var result,
+                testFn,
+                subject;
+
+            testFn = function () {
+                return this;
+            };
+
+            expect(function () {
+                result = testFn.call();
+            }).not.toThrow();
+
+            expect(result).toBe(global);
```

Comment:
in these tests, why not just `expect(testFn.call()).toBe(global)`? in what scenario would `call` ever throw on a non-throwing function?